src/articles/comments/comments.repository.ts

```diff
@@ -0,0 +1,46 @@
+import { and, desc, eq } from 'drizzle-orm';
+
+import type { Database } from '@/database.providers';
+import { articles, comments } from '@articles/articles.model';
+import type { CommentToCreate } from './comments.schema';
+
+export class CommentsRepository {
+  constructor(private readonly db: Database) {}
+
+  async create(commentData: CommentToCreate) {
+    const [comment] = await this.db
+      .insert(comments)
+      .values(commentData)
+      .returning();
+    return comment;
+  }
+
+  async findById(id: number) {
+    const result = await this.db.query.comments.findFirst({
+      where: eq(comments.id, id),
+    });
+    return result;
+  }
```

Comment:
What do you think about grabbing more data in each query? Something like this
```suggestion
   async findById(id: number) {
    const result = await this.db.query.comments.findFirst({
      where: eq(comments.id, id),
      with: {
        author: {
          columns: {
            id: true,
            username: true,
            bio: true,
            image: true,
            },
            with: {
              followers: true,
            }
          } 
        } 
    });
    return result;
  }
```

Reply:
Very useful!

Reply:
Unresolving, on second thought why did we do this again? 😅

Reply:
Can you cite a specific use case for this ?

Reply:
To avoid multiple queries to get a single comment. Currently the logic is:
 1. get a comment
 2. const authorUsername = await this.getAuthorUsername(comment.authorId);
 3. const authorProfile = await this.profilesService.findByUsername(
          currentUserId || 0,
          authorUsername,
        );

This is 3 database queries to just get a single comment. And in case of getComments, the last two query happens for every comment, which is spamming the database with queries.

Instead, you can request all relevant data at once by joining the tables, and then all that's left to do is format it to match the response schema.

To be fair, it would be even better to write a query that returns exactly what we need, but that would require hand writing sql and not using drizzle orm

Reply:
That's a great idea. I thought about it a bit. Doing the exact same logic you proposed, but for `findManyByArticleId`, instead of ``findById`, should do the trick.

Check out my latest commit to the PR and lmk if it makes you happy